content/3_Silver/Graph_Traversal.mdx

```diff
@@ -619,7 +619,60 @@ See [here.](https://codeforces.com/blog/entry/67883)
 
 ### BFS Solution
 
-<IncompleteSection />
+<CPPSection>
+
+```cpp
+#include <cstdio>
+#include <queue>
+#include <vector>
+
+const int MN = 1e5 + 10;
+
+int N, M, ans, rep[MN];
+std::vector<int> adj_list[MN];
+bool visited[MN];
+
+void bfs(int node) {
+	std::queue<int> q;
+	q.push(node);
+	visited[node] = true;
+
+	while (!q.empty()) {
+		int curr = q.front();
+		q.pop();
+
+		for (int u : adj_list[curr]) {
+			if (!visited[u]) {
+				q.push(u);
+				visited[u] = true;
+			}
+		}
+	}
+}
+
+int count_components() {
+	int count = 0;
+	for (int i = 1; i <= N; ++i)
+		if (!visited[i]) {
+			rep[count++] = i;
+			bfs(i);
+		}
+	return count;
+}
+
+int main() {
+	scanf("%d%d", &N, &M);
```

Comment:
use `cin` for consistency

Reply:
but scanf is used in the module in the dfs code.

Reply:
but `cin` is used literally everywhere else